modules/govuk_env_sync/files/govuk_env_sync.sh

```diff
@@ -683,19 +683,20 @@ nagios_code=2
 case ${action} in
   push)
     if [ "${dbms}" == "mongo" ] && [ "$(is_writable_mongo)" != "true" ]; then
-      trap - EXIT # remove the trap, as we do not want to send a nagios message in this case
       log "This machine is not a mongo master. Skipping."
-      exit
+    else
+      create_tempdir
+      create_timestamp
+      set_filename
+      "dump_${dbms}"
+      "push_${storagebackend}"
+      remove_tempdir
     fi
-    create_tempdir
-    create_timestamp
-    set_filename
-    "dump_${dbms}"
-    "push_${storagebackend}"
-    remove_tempdir
     ;;
   pull)
-    if [ "$("is_writable_${dbms}")" == 'true' ]; then
+    if [ "$("is_writable_${dbms}")" != "true" ]; then
+      log "${dbms} is not writeable. Skipping."
+    elif [ "$("is_writable_${dbms}")" == 'true' ]; then
```

Comment:
Given this is just the opposite of the `if` condition, it looks to be unnecessary?

Reply:
👍